api/api/projects_api_test.go

```diff
@@ -506,14 +633,24 @@ func TestUpdateProject(t *testing.T) {
 
 				assert.Equal(t, tC.expectedResponse.code, rr.Code)
 				if tC.expectedResponse.code >= 200 && tC.expectedResponse.code < 300 {
-					project := &models.Project{}
-					err = json.Unmarshal(rr.Body.Bytes(), &project)
-					assert.NoError(t, err)
+					switch tC.expectedResponse.data.(type) {
+					case *models.Project:
+						project := &models.Project{}
+						err = json.Unmarshal(rr.Body.Bytes(), &project)
+						assert.NoError(t, err)
 
-					project.CreatedAt = now
-					project.UpdatedAt = now
+						project.CreatedAt = now
+						project.UpdatedAt = now
 
-					assert.Equal(t, tC.expectedResponse.data, project)
+						assert.Equal(t, tC.expectedResponse.data, project)
+					case map[string]interface{}:
+						var responseMessage map[string]interface{}
+						err = json.Unmarshal(rr.Body.Bytes(), &responseMessage)
+						assert.NoError(t, err)
+						assert.Equal(t, tC.expectedResponse.data, responseMessage)
+					default:
+						t.Fatal("unexpected type for expectedResponse.data")
+					}
```

Comment:
Haha I appreciate the effort in implementing two different checks for this but a simpler way to do this (I think) is to just compare the expected json and the actual json directly (you can pass both into a string and compare them I believe). No need to change this however, just bringing this up :)